macroslib/src/cpp/fenum.rs

```diff
@@ -142,6 +144,71 @@ enum {enum_name} {{"#,
     Ok(())
 }
 
+fn generate_c_code_for_enum(
+    ctx: &mut CppContext,
+    enum_info: &ForeignEnumInfo,
+) -> std::result::Result<(), DiagnosticError> {
+    let c_path = ctx
+        .cfg
+        .output_dir
+        .join(cpp_code::c_header_name_for_enum(enum_info));
+    let mut file = FileWriteCache::new(&c_path, ctx.generated_foreign_files);
+    let enum_doc_comments = cpp_code::doc_comments_to_c_comments(&enum_info.doc_comments, true);
+
+    writeln!(
+        file,
+        r#"// Automatically generated by flapigen
+#pragma once
+
+{doc_comments}
+
+enum {enum_name} {{"#,
+        enum_name = enum_info.name,
+        doc_comments = enum_doc_comments,
+    )
+        .expect(WRITE_TO_MEM_FAILED_MSG);
+
+    for (i, item) in enum_info.items.iter().enumerate() {
+        //Enums are aligned left, so we pass true to get left aligned comments.
+        let mut doc_comments = cpp_code::doc_comments_to_c_comments(&item.doc_comments, true);
+        if !doc_comments.is_empty() && !doc_comments.ends_with('\n') {
+            doc_comments.push('\n');
+        }
+        writeln!(
+            file,
+            "{doc_comments}    {enum_name}_{item_name} = {index}{separator}",
+            item_name = item.name,
+            enum_name = enum_info.name,
+            index = i,
+            doc_comments = doc_comments,
+            separator = if i != enum_info.items.len() - 1 {
+                ",\n"
+            } else {
+                ""
+            },
+        )
+            .expect(WRITE_TO_MEM_FAILED_MSG);
+    }
+
+    // Need to name it <enum_name>Opaque, else this may conflicts with C++ bindings code.
+    writeln!(
+        file,
+        r"}};
+
+typedef enum {enum_name} {enum_name}Opaque;
```

Comment:
All looks good, except of naming (the most complex question of programming). 
"Opaque" elsewhere in crate used for some kind of [type erasure](https://en.wikipedia.org/wiki/Type_erasure). 
So "opaque" is related to that real type is invisible from `<Type>Opaque`. In case of `enum` type is visible,
so I suggest name it in some other way. For example "C{enum_name}`

Reply:
Fixed.